src/cpp/fastrtps_deprecated/publisher/PublisherHistory.cpp

```diff
@@ -290,6 +293,49 @@ bool PublisherHistory::remove_change_g(
     return remove_change_pub(a_change);
 }
 
+bool PublisherHistory::remove_instance_changes(
+        const rtps::InstanceHandle_t& handle,
+        const rtps::SequenceNumber_t& seq_up_to)
+{
+    if (mp_writer == nullptr || mp_mutex == nullptr)
+    {
+        logError(RTPS_HISTORY, "You need to create a Writer with this History before using it");
+        return false;
+    }
+
+    if (topic_att_.getTopicKind() == NO_KEY)
+    {
+        logError(RTPS_HISTORY, "Cannot be removed instance changes of a NO_KEY DataType");
+        return false;
+    }
+
+    std::lock_guard<RecursiveTimedMutex> guard(*this->mp_mutex);
+    t_m_Inst_Caches::iterator vit;
+    if (!this->find_or_add_key(handle, &vit))
```

Comment:
No need to add here.
```suggestion
    t_m_Inst_Caches::iterator vit;
    vit = keyed_changes_.find(handle);
    if (vit == keyed_changes_.end())
```